Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.cpp

```diff
@@ -118,7 +128,8 @@ void WebSpeechSynthesisClient::pause()
             observer->didPauseSpeaking();
     };
 
-    protectedPage()->sendWithAsyncReply(Messages::WebPageProxy::SpeechSynthesisPause(), WTFMove(completionHandler));
+    if (RefPtr page = m_page.get())
```

Comment:
Maybe a dumb question--why would this cause crashes? Won't the completion handler just go out of scope if we don't call it? Also, does this new version work? (Null-check page and return before the completion handler is declared instead of an else clause that calls the completion handler).

Reply:
CompletionHandler has an assertion in its destructor to make sure that it has been called. Not calling a completion handler will generally lead to hangs or breakage so the assertion makes sure we catch these cases.

Reply:
Ahhh ok that makes sense. This new iteration of the code null-check page and returns before the completion handler is created if page is null. So we won't hit the assertion now.